README.md

```diff
@@ -0,0 +1,326 @@
+# LIS3MDL Magnetometer Class
+
+This class allows the Electric Imp to drive the [LIS3MDL](http://www.st.com/web/catalog/sense_power/FM89/SC1449/PF255198) [Magnetometer](http://www.st.com/st-web-ui/static/active/en/resource/technical/document/application_note/DM00136626.pdf).  This device is a low-power, highly configurable 3-axis magnetic sensor with support for user-defined interrupts.
+
+The sensor supports I²C and SPI interfaces.  This library currently only supports the I²C interface.
+
+## Examples and Hardware
+
+For an example of this hardware integrated in a reference design, see the [Nora overview](https://electricimp.com/docs/hardware/resources/reference-designs/nora/).
+
+## Constructor: LIS3MDL(*i2c, [address]*)
+
+Creates and initializes an object representing the LIS3MDL magnetometer.  Note that this device must be enabled with a call to [`enable(state)`](#enablestate) before its sensors can be read from.
+
+The LIS3MDL will be started in continuous-conversion mode.
+
+| Parameter | Type         | Default    | Description |
+|-----------|--------------|------------|-------------|
+| i2c       | hardware.i2c | (Required) | A pre-configured [I²C object](https://electricimp.com/docs/api/hardware/i2c/) |
+| address   | Byte         | 0x1C       | The (8-bit) I²C address for the LIS3MDL |
+
+### Usage
+
+```squirrel
+#require "LIS3MDL.class.nut:1.0.0"
+
+local i2c = hardware.i2c89;
+i2c.configure(CLOCK_SPEED_400_KHZ);
+
+// Use alternate I2C address - SA1 pin tied high
+magneto <- LIS3MDL(i2c, 0x3C);
```

Comment:
A magneto is a type of electrical generator, typically used in aircraft and other things with simple internal combustion engines (https://en.wikipedia.org/wiki/Magneto). 

Can we call this something else?